src/libcore/ptr.rs

```diff
@@ -91,15 +91,50 @@ pub const fn null<T>() -> *const T { 0 as *const T }
 pub const fn null_mut<T>() -> *mut T { 0 as *mut T }
 
 /// Swaps the values at two mutable locations of the same type, without
-/// deinitializing either. They may overlap, unlike `mem::swap` which is
-/// otherwise equivalent.
+/// deinitializing either. The values pointed at by `x` and `y` may overlap,
+/// unlike `mem::swap` which is otherwise equivalent. If the values do overlap,
+/// then the overlapping region of memory will be derived from `x`.
```

Comment:
I don't think I understood the last sentence here until I looked at the example. Is there any way to write this more clearly? (The answer might be "no." I don't have any particularly good ideas.)

Reply:
another wording that comes to mind: 

>If the values do overlap, then the overlapping region of memory from `x` takes priority.

Reply:
That does actually seem better to me.

Reply:
adjusted in the latest force push

Reply:
I have to be honest, I didn't understand "takes priority" and I find the original wording better :/

Reply:
"will be used" is another possibility. It might also make sense to simply be like "see the example below" since this is definitely something an example clarifies greatly.

Reply:
applied @Gankro's suggestions in the latest force push

```diff
--- a/src/libcore/ptr.rs
+++ b/src/libcore/ptr.rs
@@ -91,9 +91,12 @@ pub const fn null<T>() -> *const T { 0 as *const T }
 pub const fn null_mut<T>() -> *mut T { 0 as *mut T }
 
 /// Swaps the values at two mutable locations of the same type, without
-/// deinitializing either. The values pointed at by `x` and `y` may overlap,
-/// unlike `mem::swap` which is otherwise equivalent. If the values do overlap,
-/// then the overlapping region of memory from `x` takes priority.
+/// deinitializing either.
+///
+/// The values pointed at by `x` and `y` may overlap, unlike `mem::swap` which
+/// is otherwise equivalent. If the values do overlap, then the overlapping
+/// region of memory from `x` will be used. This is demonstrated in the
+/// examples section below.
 ///
 /// # Safety
 ///
```